### ISSUE TITLE
Cosine Embedding Loss

### ISSUE THREAD

**Issue by varunsaboo** [pytorch/pytorch#8316]
## Issue description
Cosine Embedding Loss does not work when giving the expected and predicted tensors as batches..
Is this done intentionally?

**Comment by vishwakftw** [comment]
It works for batches of tensors. Please look at the documentation for [`CosineEmbeddingLoss`](https://pytorch.org/docs/master/nn.html#cosineembeddingloss).

Example:
```python
>>> x1 = torch.randn(3, 4)
>>> x2 = torch.randn(3, 4)
>>> y = torch.empty(3).bernoulli_().mul_(2).sub_(-1)
>>> l = torch.nn.CosineEmbeddingLoss()
>>> def ang(a, b):
...    torch.dot(a, b) / (torch.norm(a) * torch.norm(b))
...
>>> ang(x1[0], x2[0])
tensor(1.00000e-02 *
       9.8725)
>>> ang(x1[1], x2[1])
tensor(1.00000e-02 *
       7.6634)
>>> ang(x1[2], x2[2])
tensor(-0.4407)
>>> # By definition: the loss value should be mean of (1 - ang(x1[0], x2[0])), ang(x1[1], x2[1]) and 0
>>> # Which is approx 0.326
>>> l(x1, x2, y).item() # prints 0.325969
```

**Comment by varunsaboo** [comment]
According to the documentation, y is supposed to be 1 or -1 to indicate whether the two vectors are supposed to be similar or not. Why are you creating a distribution of y instead? Shouldn't you pass a tensor containing only 1's and -1's?

**Comment by vishwakftw** [comment]
```y = torch.empty(3).bernoulli_().mul_(2).sub_(1)``` does exactly that. The behaviour is defined as per the documentation.

**Comment by zou3519** [comment]
Closing this issue because it doesn't look like a bug report or feature request. @varunsaboo please give us a code example (with relevant output and what you expected) if you do find a bug!

**Comment by gauravkoradiya** [comment]
if u make vector of y as one by using torch.ones(dim) than it gives a perfect output.

**Comment by cristianoBY** [comment]
yes, I agree with what @gauravkoradiya said! use `y = torch.ones(dim)` for similar and `y = -torch.ones(dim)` for dissimilar. I am a little confused with @vishwakftw 's example of generating a tensor with random 1 and -1. Does this separately compute the cosine loss across each row of the tensor? Anyway, in the doc, I did not see how to specify the dimension for computing the loss.
